scripts/test-e2e-local.js

```diff
@@ -0,0 +1,201 @@
+// breaks a bunch of stuff. This should be enough.
+// TODO: follow up with Nicola in understanding the impact of the commit, and if we can remove this
+// also, remove the folder
+process.env.TMP_PUBLISH_DIR = fs.mkdtempSync(
+  path.join(os.tmpdir(), 'rn-publish-'),
+);
+console.info(`The temp folder is ${process.env.TMP_PUBLISH_DIR}`);
+
+const argv = yargs
+  .option('t', {
+    alias: 'target',
+    default: 'RNTester',
+    choices: ['RNTester', 'RNTestProject'],
+  })
+  .option('p', {
+    alias: 'platform',
+    default: 'iOS',
+    choices: ['iOS', 'Android'],
+  })
+  .option('h', {
+    alias: 'hermes',
+    type: 'boolean',
+    default: true,
+  }).argv;
+
+/*
+ * see the test-local-e2e.js script for clean up process
+ */
+
+// command order: we ask the user to select if they want to test RN tester
+// or RNTestProject
+
+// if they select RN tester, we ask if iOS or Android, and then we run the tests
+// if they select RNTestProject, we run the RNTestProject test
+
+// let's check if Metro is already running, if it is let's kill it and start fresh
+if (isPackagerRunning() === 'running') {
+  exec("lsof -i :8081 | grep LISTEN | /usr/bin/awk '{print $2}' | xargs kill");
+}
+
+if (argv.target === 'RNTester') {
+  //FIXME: make sure that the commands retains colors
+  // (--ansi) doesn't always work
+  // see also https://github.com/shelljs/shelljs/issues/86
+
+  if (argv.platform === 'iOS') {
+    console.info("We're going to test the Hermes version of RNTester iOS");
+    exec(
+      `cd packages/rn-tester && USE_HERMES=${
+        argv.hermes ? 1 : 0
+      } bundle exec pod install --ansi`,
+    );
+
+    // if everything succeeded so far, we can launch Metro and the app
+    // start the Metro server in a separate window
+    launchPackagesInSeparateWindow();
+
+    // launch the app on iOS simulator
+    pushd('packages/rn-tester');
+    exec('npx react-native run-ios --scheme RNTester');
+    popd();
+  } else {
+    // we do the android path here
+
+    launchAndroidEmulator();
+
+    if (argv.hermes) {
+      console.info(
+        "We're going to test the Hermes version of RNTester Android",
+      );
+      exec(
+        './gradlew :packages:rn-tester:android:app:installHermesDebug --quiet',
+      );
+    } else {
+      console.info("We're going to test the JSC version of RNTester Android");
+      exec('./gradlew :packages:rn-tester:android:app:installJscDebug --quiet');
+    }
+
+    // if everything succeeded so far, we can launch Metro and the app
+    // start the Metro server in a separate window
+    launchPackagesInSeparateWindow();
+    // just to make sure that the Android up won't have troubles finding the Metro server
+    exec('adb reverse tcp:8081 tcp:8081');
+    // launch the app
+    exec(
+      'adb shell am start -n com.facebook.react.uiapp/com.facebook.react.uiapp.RNTesterActivity',
```

Comment:
Idea for future self: it should be possible to `yarn android` inside `packages/rn-tester`

Reply:
it might work actually, let me give it a try...

Reply:
ok so for the future self; I tried replacing this with

```
    // launch the app on Android simulator
    pushd('packages/rn-tester');
    exec('npx react-native run-android');
    popd();
```

and it fails with `error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.`

so there's some custom stuff in rn-tester folder that we'd need to address for this to work. I think we can leave it for a future task

Reply:
> so there's some custom stuff in rn-tester folder that we'd need to address for this to work. I think we can leave it for a future task

Yeah, something to do in the future :+1: